modules/boot-commons/src/main/java/org/mule/runtime/module/boot/commons/internal/MuleContainerWrapperProvider.java

```diff
@@ -35,6 +38,12 @@ public static MuleContainerWrapper getMuleContainerWrapper() {
     return INSTANCE;
   }
 
+  @Override
+  public MuleContainerLifecycleWrapper getMuleContainerLifecycleWrapper() {
+    getMuleContainerWrapper();
+    return INSTANCE;
+  }
```

Comment:
This shouldn't be necessary, just change `getMuleContainerWrapper` return type from `MuleContainerWrapper` to  `MuleContainerLifecycleWrapper`.